projects/angular/src/button/button-group/button.ts

```diff
@@ -148,6 +167,12 @@ export class ClrButton implements LoadingListener {
     this._click.emit(true);
   }
 
+  ngOnInit() {
+    if (this.inMenu && !this.id) {
+      this.id = uniqueIdFactory();
```

Comment:
I think this can just be initialized at the `_id` property declaration. That simplifies the code and could prevent an extra DOM update.

Reply:
It should be okay to just initialize the id of all buttons, not just the ones in the in the menu.

Reply:
I'll have to change some of the tests for button because they are validating if id is '' if it's not set.

Reply:
Oh. An empty `id` is invalid anyway...